### ISSUE TITLE
ALL_TOKEN but only for prunable layers (ConvNd and Linear)

### ISSUE THREAD

**Issue by eldarkurtic** [neuralmagic/sparseml#139]
**Is your feature request related to a problem? Please describe.**
When one wants to prune all prunable layers of a model, it is necessary to specify all layer names manually in the YAML recipe. This is not convenient, especially when starting to work with new models, one has to make a dummy run to extract layer names and to use them later on to create a YAML config. As far as I understand, there is `ALL_TOKEN = '__ALL__'` which adds all layers (including BatchNorm, biases and so on). Most of the time, we don't want to prune all of these. 

**Describe the solution you'd like**
To be able to create a YAML file with: `params: __ALL_PRUNABLE__`

**Describe alternatives you've considered**
My current solution is to do something like this in the `GMPruningModifier` class:
```python
if '__ALL_PRUNABLE__' in self._params:
            self._params = [layer_name + ".weight" for (layer_name, _) in get_prunable_layers(module)]
```

**Comment by bfineran** [comment]
Hi @eldarkurtic thank you for the feature request.  Our pruning modifiers do support regex for parameter matching if you prefix them with `"re:"`.  So, to match all convs for example, you could do something like `"re:.*conv.weight` depending on your parameter names.  Could this satisfy your feature request?

**Comment by eldarkurtic** [comment]
Hi :) 
I think that regex might not be the best solution for this feature, since it relies on the fact that `ConvNd` layers will actually have `conv` in their name. And this highly depends on the structure of the model/checkpoint that we are using. For example, Resnet50 model has `layerX.downsample.0.weight` which is a `Conv` layer without `conv` in the name. I guess the same can happen for `Linear` layers, since most of the time the last `Linear` layer is called `fc`.

**Comment by markurtz** [comment]
Hey @eldarkurtic, definitely seems like a reasonable ask. Feel free to put up a diff if you'd like and we'll review, otherwise we'll be sure to add it to the current on going sprint for expansion of pruning capabilities in PyTorch.

Another note is that we're also working on functionality and apis to create base and automatically optimized recipes, so ideally that becomes the more preferred pathway in the near future. It still automates everything away, but makes the results easily readable and editable as well.

**Comment by markurtz** [comment]
Closing this issue out now as support has now been added in on the PyTorch side
